agent/wsclient/client.go

```diff
@@ -218,14 +231,21 @@ func (cs *ClientServerImpl) MakeRequest(input interface{}) error {
 
 	// Over the wire we send something like
 	// {"type":"AckRequest","message":{"messageId":"xyz"}}
-	return cs.Conn.WriteMessage(websocket.TextMessage, send)
+	return cs.MakeSyncRequest(send)
+}
+
+// MakeSyncRequest wraps the low level websocket write method with a lock
+func (cs *ClientServerImpl) MakeSyncRequest(send []byte) error {
```

Comment:
I think I'd keep the name as `WriteMessage` and document that it's overriding the `conn.WriteMessage` to force synchronous behavior in the godoc comment.